scipy/stats/_continuous_distns.py

```diff
@@ -3490,12 +3490,17 @@ def _rvs(self, mu, size=None, random_state=None):
         return random_state.wald(mu, 1.0, size=size)
 
     def _pdf(self, x, mu):
-        # invgauss.pdf(x, mu) =
-        #                  1 / sqrt(2*pi*x**3) * exp(-(x-mu)**2/(2*x*mu**2))
-        return 1.0/np.sqrt(2*np.pi*x**3.0)*np.exp(-1.0/(2*x)*((x-mu)/mu)**2)
+        return np.exp(self._logpdf(x, mu))
 
     def _logpdf(self, x, mu):
-        return -0.5*np.log(2*np.pi) - 1.5*np.log(x) - ((x-mu)/mu)**2/(2*x)
+        a = -0.5 * np.log(2 * np.pi) - 1.5 * np.log(x)
+        out = _lazywhere(
+            mu == np.inf,
```

Comment:
You can use `np.isinf` here

Reply:
`np.isposinf`?

```
In [4]: np.isinf(-float('inf'))
Out[4]: True
```

Reply:
If only pos inf is the use case then yes probably that's safer